libs/logging/src/main/java/org/elasticsearch/logging/Level.java

```diff
@@ -0,0 +1,154 @@
+     */
+    public static final Level ERROR = new Level("ERROR", StandardLevels.ERROR);
+    /**
+     * An event that might possible lead to an error.
+     */
+    public static final Level WARN = new Level("WARN", StandardLevels.WARN);
+    /**
+     * An event for informational purposes.
+     */
+    public static final Level INFO = new Level("INFO", StandardLevels.INFO);
+    /**
+     * A general debugging event.
+     */
+    public static final Level DEBUG = new Level("DEBUG", StandardLevels.DEBUG);
+    /**
+     * A fine-grained debug message, typically capturing the flow through the application.
+     */
+    public static final Level TRACE = new Level("TRACE", StandardLevels.TRACE);
+    /**
+     * All events should be logged.
+     */
+    public static final Level ALL = new Level("ALL", StandardLevels.ALL);
+
+    private static final ConcurrentMap<String, Level> LEVELS = new ConcurrentHashMap<>();
+
+    static {
+        LEVELS.put(OFF.name, OFF);
+        LEVELS.put(FATAL.name, FATAL);
+        LEVELS.put(ERROR.name, ERROR);
+        LEVELS.put(WARN.name, WARN);
+        LEVELS.put(INFO.name, INFO);
+        LEVELS.put(DEBUG.name, DEBUG);
+        LEVELS.put(TRACE.name, TRACE);
+        LEVELS.put(ALL.name, ALL);
+    }
+
+    private final String name;
+    private final int severity;
+
+    private Level(String name, int severity) {
+        this.name = name;
+        this.severity = severity;
+    }
+
+    /**
+     * Retrieves an existing Level or creates on if it didn't previously exist.
+     *
+     * @param name The name of the level.
+     * @param severity The integer value for the Level. If the level was previously created this value is ignored.
+     * @return The Level.
+     * @throws java.lang.IllegalArgumentException if the name is null or intValue is less than zero.
+     */
+    public static Level forName(String name, int severity) {
+        Objects.requireNonNull(name);
+
+        final String levelName = name.trim().toUpperCase(Locale.ROOT);
+        var level = new Level(levelName, severity);
+        if (LEVELS.putIfAbsent(levelName, level) != null) {
+            return LEVELS.get(levelName);
+        }
+        return level;
+    }
+
+    /**
+     * Return the Level associated with the name.
+     *
+     * @param name The name of the Level to return.
+     * @return The Level.
+     * @throws java.lang.NullPointerException if the Level name is {@code null}.
+     * @throws java.lang.IllegalArgumentException if the Level name is not registered.
+     */
+    public static Level forName(final String name) {
+        Objects.requireNonNull(name);
+        final String levelName = name.trim().toUpperCase(Locale.ROOT);
+        final Level level = LEVELS.get(levelName);
+        if (level != null) {
+            return level;
+        }
+        throw new IllegalArgumentException("Unknown level constant [" + levelName + "].");
+    }
+
+    public static Collection<Level> values() {
```

Comment:
we don't have usages of org.apache.logging.log4j.Level#values in our codebase. Unlikely that plugins have too.
I only use this in testing. The only usages I could think of is to prevent exception when trying to get Level.forName
```
if(Level.values().contains("CustomName")) 
Level.forName("CustomName")

Reply:
removed, because Level is now an enum